### PR TITLE
KAA-1573: [C++] No linking to threading library on Intel Edison platform

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @Slash32 @vadimol

### CODE REVIEW

**Review by Slash32 — APPROVED** [review]

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -133,7 +133,7 @@
 #       Default: `0`.
 
 cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
-project(KaaCppSdk CXX)
+project(KaaCppSdk C CXX)
```
Comment (vadimol) [inline client/client-multi/client-cpp/CMakeLists.txt:136]: Is it necessary?

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 2]
```diff
@@ -393,6 +393,7 @@ endif()
 find_package(Boost 1.54 REQUIRED COMPONENTS log system thread)
 find_package(Avro REQUIRED)
 find_package(Botan REQUIRED)
+find_package(Threads REQUIRED)
```
Comment (vadimol) [inline client/client-multi/client-cpp/CMakeLists.txt:396]: Have you investigated, why the
_find_package(Boost 1.54 REQUIRED COMPONENTS log system **thread**)_
cannot find this library on Intel Edison?

Comment (RostakaGmfun) [reply client/client-multi/client-cpp/CMakeLists.txt:396]: Indeed, the `find_package(Boost 1.54 REQUIRED COMPONENTS log system thread)` successfully finds the **boost.threads** library, but don't confuse it with **Threads** package which provides platform-specific threading implementation. Initially it was expected that the above command will take care of finding the Threads package and initialize the `CMAKE_THREAD_LIBS_INIT` CMake variable, however this does not seem to be true for all platforms.

File: client/client-multi/client-cpp/CMakeLists.txt (hunk 1 above)
Comment (RostakaGmfun) [reply client/client-multi/client-cpp/CMakeLists.txt:136]: Yep, the `find_package(Threads)` fails if the project does not contain the C language.

**Review by vadimol — APPROVED** [review]

**Review by rasendubi — APPROVED** [review]

**Pull request merged**